components/x-gift-article/src/lib/highlightsApi.js

```diff
@@ -0,0 +1,56 @@
+export default class HighlightsApiClient {
+	constructor(baseUrl) {
+		this.baseUrl = baseUrl
+	}
+
+	/**
+	 * Concatenates protocol, domain and path URLs.
+	 * @param {string} path URL Path
+	 * @returns {string} Fetch URL
+	 * @throws {Error} if baseURL is empty
+	 */
+	getFetchUrl(path) {
+		if (!this.baseUrl) {
+			throw new Error('User Annotations API base url missing')
+		}
+
+		return `${this.baseUrl}${path}`
+	}
+
+	/**
+	 * Makes a fetch request to the path with additional options
+	 * @param {string} path URL path
+	 * @param {RequestInit} additionalOptions fetch additional options
+	 * @returns {Promise<object>} A promise that resolves to the requested URL response parsed from json
+	 */
+	async fetchJson(path, additionalOptions) {
+		const url = this.getFetchUrl(path)
+		const options = Object.assign(
+			{
+				credentials: 'include',
+				headers: {
+					'Content-Type': 'application/json'
+				}
+			},
+			additionalOptions
+		)
+
+		const response = await fetch(url, options)
+		return await response.json()
```

Comment:
**Non-blocking comment**: I'd prefer to wrap `await response.json()` in a variable and then return it.
```
const output = await response.json()
return output
```

Reply:
we can do that but i don't think that adding a new variable for the sake of just returning it's value is best , since this code does the same without adding it